canonical/karma-k8s-operator
# Add TLS support

## Issue
Need to have karma communicating with alertmanager over TLS.


## Solution
Add TLS support to charm.

Drive-by fixes:
- Fixes #29.
  - Drop `external_hostname` and enable `stripPrefix`.
- Adjust pydantic model so we can pin "pydantic<2" like the rest of the libs.



## Context



## Testing Instructions
1. Relate avalanche-prometheus-alertmanager-karma, and make sure you see the watchdog alert in karma.
2. Relate prometheus, alertmanager, karma to self-signed-certificates and make sure you still see the alert.


## Release Notes
Add TLS support.
